soccer/include/rj_vision_filter/VisionFilter.hpp

```diff
@@ -63,15 +89,62 @@ class VisionFilter {
      */
     void GetFrames();
 
-    std::thread worker;
+    /**
+     * @brief Returns the team angle (It's horrible, but it's only temporary
+     * until VisionFilter gets refactored).
+     * @return The team angle.
+     */
+    [[nodiscard]] double TeamAngle() const {
+        const bool defend_plus_x = config_client_.gameSettings().defend_plus_x;
+        return defend_plus_x ? -M_PI_2 : M_PI_2;
+    }
+
+    /**
+     * @brief Returns the transform from the world to the team.
+     * @return The transform from the world to the team frame.
+     */
+    [[nodiscard]] Geometry2d::TransformMatrix WorldToTeam() const {
+        Geometry2d::TransformMatrix world_to_team =
+            Geometry2d::TransformMatrix::translate(
+                0, config_client_.fieldDimensions().length / 2.0f);
+        world_to_team *= Geometry2d::TransformMatrix::rotate(
+            static_cast<float>(TeamAngle()));
+        return world_to_team;
+    }
 
-    std::mutex worldLock;
+    mutable std::mutex world_mutex;
```

Comment:
If you merge the callback groups, I think you no longer need this. I might just be misunderstanding how those work though.

Reply:
Okay sure. I merged them.